src/GraphQL/Extensions/SchemaExtensions.cs

```diff
@@ -81,12 +84,23 @@ public static void RegisterTypes<TSchema>(this TSchema schema, params IGraphType
         /// </summary>
         /// <typeparam name="TClrType">The CLR property type from which to infer the GraphType.</typeparam>
         /// <typeparam name="TGraphType">Inferred GraphType.</typeparam>
-        public static void RegisterTypeMapping<TClrType, TGraphType>(this ISchema schema)
+        public static void RegisterTypeMapping<TClrType,
+#if NET5_0_OR_GREATER
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)]
+#endif
+        TGraphType>(this ISchema schema)
             where TGraphType : IGraphType
         {
+            Preserve<GraphQLClrInputTypeReference<TClrType>>();
+            Preserve<GraphQLClrOutputTypeReference<TClrType>>();
```

Comment:
hints

Reply:
I think these uses of `Preserve` are unavoidable.  Since it is a generic type, there is no type that can be added to an attribute at compile time.  However, the AOT trimmer can follow the type and understand the proper type to preserve.

Note that this may not cover every scenario - for instance, `ListGraphType<GraphQLClrOutputTypeReference<TClrType>>` may not work.  We could add the six basic combinations (none/nonnull/list/nonnull list/list nonnull/nonnull list nonnull)... I just put in the two basic ones.  Someone can always explicitly preserve additional types if needed.

Note that if someone is calling this method, they MUST be expecting to use some kind of type reference.  Otherwise there would be no reason to call this method.  So it seems logical to preserve at minimum the two standard references within this method.

Reply:
> I think these uses of Preserve are unavoidable. Since it is a generic type, there is no type that can be added to an attribute at compile time.

Even if use https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/prepare-libraries-for-trimming?source=recommendations#dynamicdependency with opened generic? I did not try myself.

Reply:
OK. I understand now the purpose of hints here.